drivers/clock_control/clock_control_r8a779f0_cpg_mssr.c

```diff
@@ -116,22 +162,84 @@ int r8a779f0_cpg_mssr_start_stop(const struct device *dev, clock_control_subsys_
 
 static uint32_t r8a779f0_get_div_helper(uint32_t reg_val, uint32_t module)
 {
+	uint32_t divider = RCAR_CPG_NONE;
+
 	switch (module) {
 	case R8A779F0_CLK_S0D12_PER:
 	case R8A779F0_CLK_CL16M:
 		return 1;
+	case CLK_SDSRC:
+		reg_val >>= R8A779F0_CLK_SDSRC_DIV_SHIFT;
+		reg_val &= R8A779F0_CLK_SDSRC_DIV_MASK;
+		/*  setting of 3 is prohibited */
```

Comment:
> The comment contradicts to the check below. Should it be "setting a value less than 3 is prohibited" ?

After masking reg_val, we can have only values in the range 0-3. Therefore, the statements 'setting of 3 is prohibited' and 'setting a value less than 3 is prohibited' are equivalent in this context

> Also, should you somehow warn the user about the invalid value?

I'll add warn prints here and below

Reply:
> After masking reg_val, we can have only values in the range 0-3. Therefore, the statements 'setting of 3 is prohibited' and 'setting a value less than 3 is prohibited' are equivalent in this context

Technically yes, but it depends on knowledge of what exact value of `R8A779F0_CLK_SDSRC_DIV_MASK` is. For someone reading the code it is not immediately clear why comment says that "3 is prohibited", but actual code checks for value less than 3. Actually, this is what happened to me and I expect that the same will happen to others.

It would be really helpful to extend comment to something like "After masking possible values are in range 0-3, but HW specification forbids value 3".

Reply:
fixed